File: app/enums/clinical_level.py
from enum import StrEnum


class ClinicalLevel(StrEnum):
    FAR = "FAR"
    EFR = "EFR"
    EMT = "EMT"
    PARAMEDIC = "PARAMEDIC"
    ADVANCED_PARAMEDIC = "ADVANCED_PARAMEDIC"

    def label(self) -> str:
        return {
            ClinicalLevel.FAR: "FAR",
            ClinicalLevel.EFR: "EFR",
            ClinicalLevel.EMT: "EMT",
            ClinicalLevel.PARAMEDIC: "Paramedic",
            ClinicalLevel.ADVANCED_PARAMEDIC: "Advanced Paramedic",
        }[self]

    @property
    def rank(self) -> int:
        return {
            ClinicalLevel.FAR: 1,
            ClinicalLevel.EFR: 2,
            ClinicalLevel.EMT: 3,
            ClinicalLevel.PARAMEDIC: 4,
            ClinicalLevel.ADVANCED_PARAMEDIC: 5,
        }[self]

File: app/Enums/ClinicalLevel.php
<?php

declare(strict_types=1);

namespace App\Enums;

enum ClinicalLevel: string
{
    case EFR = 'efr';
    case EMT = 'emt';
    case Paramedic = 'paramedic';
    case AdvancedParamedic = 'advanced_paramedic';
    case Nurse = 'nurse';
    case Doctor = 'doctor';

    public function label(): string
    {
        return match ($this) {
            self::EFR => 'EFR',
            self::EMT => 'EMT',
            self::Paramedic => 'Paramedic',
            self::AdvancedParamedic => 'Advanced Paramedic',
            self::Nurse => 'Nurse',
            self::Doctor => 'Doctor',
        };
    }

    public function rank(): int
    {
        return match ($this) {
            self::EFR => 1,
            self::EMT => 2,
            self::Paramedic => 3,
            self::AdvancedParamedic => 4,
            self::Nurse => 5,
            self::Doctor => 6,
        };
    }
}
